common/src/main/java/com/wynntils/utils/wynn/ContainerUtils.java

```diff
@@ -69,4 +72,16 @@ public static void clickOnSlot(int clickedSlot, int containerId, int mouseButton
     public static void closeContainer(int containerId) {
         McUtils.sendPacket(new ServerboundContainerClosePacket(containerId));
     }
+
+    public static Stream<SlotPos> getSlots(ChestMenu chestMenu) {
+        return chestMenu.slots.stream().map(SlotPos::of);
+    }
+
+    public record SlotPos(ItemStack itemStack, int x, int y) {
```

Comment:
It just feels odd to have an ItemStack here. Look at how Vanilla implements `Slot.getItem()`:
`return this.container.getItem(this.slot);`

You should do the same. Construct this with a container instead. I would also recommend renaming x and y to "row" and "column" instead, since it is not about graphic coordinates.

Reply:
Actually, the entire type seems wrong. All you really need is to convert a slot.index to row and column values. You can send in a Slot to the filter predicate, and have ContainerUtils methods "getSlotRow" and "getSlotColumn" to extract these values using % and / 9.